### ISSUE TITLE
Move YdbRepository start behavior from YdbRepository constructor to separate method

### ISSUE THREAD

**Issue by lavrukov** [ydb-platform/yoj-project#58]
YdbRepository in the constructor establishes a connection to the database, because it is impossible to separate the stages of object initialization and start.

It seems that what is needed here is a factory object which will create a SessionManager and return a Repository

**Comment by nvamelichev** [comment]
:zany_face: This is an unexpected change in YDB SDK v2, constructing a `GrpcTransport` immediately establishes a GRPC connection:
```java
    public GrpcTransport build() {
        YdbTransportImpl impl = new YdbTransportImpl(this);
        try {
            impl.init();
```

We might just add a Guava `Suppliers.memoize()` for `GrpcTransport` and `SessionManager` to make initialization lazy, as it was in repository-ydb-v1.

**Comment by nvamelichev** [comment]
:arrow_right: See PR with lazy initialization here:
https://github.com/ydb-platform/yoj-project/pull/61

**Comment by nvamelichev** [comment]
Fixed, the fix will appear in 2.2.11.
